adf/parameters.py

```diff
@@ -0,0 +1,14 @@
+"""
+Contains functionality to get parameters from ADF
+"""
+
+from py4j.protocol import Py4JJavaError
+
+def get(parameter_name, default_value, dbutils):
+    """Returns parameter from ADF"""
+    try:
+        value = dbutils.widgets.get(parameter_name)
```

Comment:
Hvis man kun har `dbutils.widgets.get("key")`, så kan den ikke få fat i parameteret fra ADF, hvis ikke man har oprettet widget først på følgende måde:
`dbutils.widgets.text("key", "", "")`

Før du laver en .get, så skal du lave:
`dbutils.widgets.text(parameter_name, default_value, "")`

Reply:
Fra deres doc:

> dbutils.widgets provides utilities for working with notebook widgets. You can create different types of widgets and get their bound value. For more info about a method, use dbutils.widgets.help("methodName").
combobox(name: String, defaultValue: String, choices: Seq, label: String): void -> Creates a combobox input widget with a given name, default value and choices
dropdown(name: String, defaultValue: String, choices: Seq, label: String): void -> Creates a dropdown input widget a with given name, default value and choices
get(name: String): String -> Retrieves current value of an input widget
getArgument(name: String, optional: String): String -> (DEPRECATED) Equivalent to get
multiselect(name: String, defaultValue: String, choices: Seq, label: String): void -> Creates a multiselect input widget with a given name, default value and choices
remove(name: String): void -> Removes an input widget from the notebook
removeAll: void -> Removes all widgets in the notebook
text(name: String, defaultValue: String, label: String): void -> Creates a text input widget with a given name and default value

https://docs.databricks.com/notebooks/widgets.html#widget-types

Reply:
Jeg har lige testet det, og det virker fint (efter jeg fik rettet min except). Man kan godt læse værdien, hvis parameteren findes i ADF. Hvis den ikke findes, så bliver den sat til default værdien.
Det er egentlig bare fordi jeg bedre kan lide at sætte værdien i koden i stedet for i en widget tekstboks.